src/XMLexport.cpp

```diff
@@ -39,619 +40,795 @@ using namespace std;
+        writeTextElement( "mLightMagenta", pHost->mLightMagenta.name() );
+        writeTextElement( "mWhite", pHost->mWhite.name() );
+        writeTextElement( "mLightWhite", pHost->mLightWhite.name() );
+        writeTextElement( "mDisplayFont", pHost->mDisplayFont.toString() );
+        writeTextElement( "mCommandLineFont", pHost->mCommandLineFont.toString() );
+        // There was a mis-spelt duplicate commandSeperator above but it is now gone
+        writeTextElement( "mCommandSeparator", pHost->mCommandSeparator );
+        writeTextElement( "commandLineMinimumHeight", QString::number(pHost->commandLineMinimumHeight) );
+
+        writeTextElement( "mFgColor2", pHost->mFgColor_2.name() );
+        writeTextElement( "mBgColor2", pHost->mBgColor_2.name() );
+        writeTextElement( "mBlack2", pHost->mBlack_2.name() );
+        writeTextElement( "mLightBlack2", pHost->mLightBlack_2.name() );
+        writeTextElement( "mRed2", pHost->mRed_2.name() );
+        writeTextElement( "mLightRed2", pHost->mLightRed_2.name() );
+        writeTextElement( "mBlue2", pHost->mBlue_2.name() );
+        writeTextElement( "mLightBlue2", pHost->mLightBlue_2.name() );
+        writeTextElement( "mGreen2", pHost->mGreen_2.name() );
+        writeTextElement( "mLightGreen2", pHost->mLightGreen_2.name() );
+        writeTextElement( "mYellow2", pHost->mYellow_2.name() );
+        writeTextElement( "mLightYellow2", pHost->mLightYellow_2.name() );
+        writeTextElement( "mCyan2", pHost->mCyan_2.name() );
+        writeTextElement( "mLightCyan2", pHost->mLightCyan_2.name() );
+        writeTextElement( "mMagenta2", pHost->mMagenta_2.name() );
+        writeTextElement( "mLightMagenta2", pHost->mLightMagenta_2.name() );
+        writeTextElement( "mWhite2", pHost->mWhite_2.name() );
+        writeTextElement( "mLightWhite2", pHost->mLightWhite_2.name() );
+        writeTextElement( "mSpellDic", pHost->mSpellDic );
+        // TODO: Consider removing these sub-elements that duplicate the same
+        // attributes - which WERE bugged - when we update the XML format, must leave
+        // them in place for now even though we no longer use them for compatibility
+        // with older version of Mudlet
+        writeTextElement( "mLineSize", QString::number(pHost->mLineSize, 'f', 1) );
+        writeTextElement( "mRoomSize", QString::number(pHost->mRoomSize, 'f', 1) );
+
+        writeEndElement(); // </Host>
+    }
+
+    writeEndElement(); // </HostPackage>
+
+    if( hasError() ) {
+        isOk = false;
+    }
+
+    // Use if() to block each XXXXPackage element to limit scope of iterator so
+    // we can use more of the same code in each block - and to escape quicker on
+    // error...
+    if( isOk ) {
+        writeStartElement("TriggerPackage");
+        for( auto it = pHost->mTriggerUnit.mTriggerRootNodeList.begin(); isOk && it != pHost->mTriggerUnit.mTriggerRootNodeList.end(); ++it ) {
+            if( ! (*it) || (*it)->mModuleMember) {
```

Comment:
Here and elsewhere in similar code I got rid of a temporary variable (which had to be of the specified type) so we could take more advantage of the C11 `auto` keyword - minor niggle is the current Qt IDE does not detect the type for members/methods used on the auto'ed variable to do the correct highlighting for them! :angel: